src/renderers/shared/reconciler/ReactMultiChild.js

```diff
@@ -276,8 +292,13 @@ var ReactMultiChild = {
      */
     _updateChildren: function(nextNestedChildrenElements, transaction, context) {
       var prevChildren = this._renderedChildren;
+      var removedChildren = {};
```

Comment:
Text components is the only thing that doesn't have children. I mean divs/Views that still implement ReactMultiChild but don't currently have any children.

In that case, flattenChildren returns null.